scanner/sbom/sbom.go

```diff
@@ -1,48 +1,43 @@
 package sbom
 
 import (
+	"bytes"
 	"context"
-	"encoding/json"
 	"fmt"
-	"time"
 
 	"github.com/quay/claircore"
+	"github.com/quay/claircore/sbom/spdx"
 	"github.com/stackrox/rox/scanner/internal/version"
 )
 
 type SBOMer struct {
 }
 
+type Options struct {
+	Name      string
+	Namespace string
+	Comment   string
+}
+
 func NewSBOMer() *SBOMer {
 	return &SBOMer{}
 }
 
-func (s *SBOMer) GetSBOM(ctx context.Context, ir *claircore.IndexReport, name, id string) ([]byte, error) {
-	// TODO(ROX-27145): remove static response and use claircore to create SBOM.
-	// Start: temporary static response
-	fakeResp := struct {
-		Msg             string   `json:"msg"`
-		Name            string   `json:"name"`
-		DocumentComment string   `json:"documentComment"`
-		Creators        []string `json:"creators"`
-		NumPackages     int      `json:"pkgs"`
-		NumDists        int      `json:"dists"`
-		NumRepos        int      `json:"repos"`
-		NumEnvs         int      `json:"envs"`
-	}{
-		Msg:             fmt.Sprintf("This fake response generated from Scanner V4 matcher on %q", time.Now().Format(time.RFC3339)),
-		Name:            id,
-		DocumentComment: fmt.Sprintf("Tech Preview - generated for '%s'", name),
-		Creators: []string{
-			fmt.Sprintf("Tool: scanner-v4-matcher-%s", version.Version),
-		},
-		NumPackages: len(ir.Packages),
-		NumDists:    len(ir.Distributions),
-		NumRepos:    len(ir.Repositories),
-		NumEnvs:     len(ir.Environments),
+func (s *SBOMer) GetSBOM(ctx context.Context, ir *claircore.IndexReport, opts *Options) ([]byte, error) {
```

Comment:
need to `nil` check the `opts`

Reply:
This would be more defensive, I figured it wasn't necessary since the only use case in the matcher service API we always populate it.

Will add the nil check for future benefit.

Reply:
https://github.com/stackrox/stackrox/pull/14135/commits/b6032f7b6665325b319db6d3529e447d911765a2